src/Installer.php

```diff
@@ -255,6 +316,38 @@ private function downloadFile(HttpClient $httpClient, $url, $targetPath)
     }
 
     /**
+     * Downloads the available puli versions.
+     *
+     * @param HttpClient $httpClient The client to use.
+     * @param string     $url        The URL to download.
+     *
+     * @return array|null The available versions, null if the download failed.
+     *
+     * @throws RuntimeException If an error occurs.
+     */
+    public function downloadVersions(HttpClient $httpClient, $url)
+    {
+        ErrorHandler::register();
+
+        $versions = $httpClient->download($url);
+
+        if (ErrorHandler::hasErrors()) {
```

Comment:
Yes that would be the easiest solution, but it felt bad because we also have an errorhandler registered